lib/sidekiq/logging.rb

```diff
@@ -5,31 +5,56 @@
 
 module Sidekiq
   module Logging
+    $stderr.puts("**************************************************")
+    $stderr.puts("⛔️ WARNING: Sidekiq::Logging module will be refactored to Sidekiq::Logger class and Sidekiq::Logging.job_hash_context will be moved to Sidekiq::JobLogger in Sidekiq 6.0. Please ensure your logging customizations are updated accordingly. ")
+    $stderr.puts("**************************************************")
 
     class Pretty < Logger::Formatter
-      SPACE = " "
-
-      # Provide a call() method that returns the formatted message.
       def call(severity, time, program_name, message)
-        "#{time.utc.iso8601(3)} #{::Process.pid} TID-#{Sidekiq::Logging.tid}#{context} #{severity}: #{message}\n"
+        "#{time.utc.iso8601(3)} #{::Process.pid} TID-#{Sidekiq::Logging.tid}#{format_context(Sidekiq::Logging.context)} #{severity}: #{message}\n"
       end
 
-      def context
-        c = Thread.current[:sidekiq_context]
-        " #{c.join(SPACE)}" if c && c.any?
+      private
+
+      def format_context(context)
+        ' ' + context.join(' ') if context.any?
```

Comment:
I think the context should format as "class=FooWorker tid=12345abc" in JSON.  In other words, adding context is just adding more key/value pairs. The JSON output may include those keys while the plain logger may omit certain keys (especially if we want to keep the same plain log output format).

Reply:
Or do we leave the context as an array of strings internally and concatenated with spaces string externally? I mean we can format context per formatter, but what should be considered as internal data structure – array of strings or hash with string keys/values?

Reply:
The solid context string is not very structured, but as far as we don't fully control the context I think it's a wise decision. That means we leave user to provide `key=value` strings to the context. Correct me if I'm wrong?